src/lib/components/Swap/SwapButton.tsx

```diff
@@ -88,26 +89,28 @@ export default function SwapButton({ disabled }: SwapButtonProps) {
     })
   }, [addTransaction, handleApproveOrPermit])
 
+  const { type: wrapType, callback: wrapCallback, error: wrapError, loading: wrapLoading } = useWrapCallback()
+
   const disableSwap = useMemo(
     () =>
       disabled ||
       !chainId ||
+      wrapLoading ||
+      (wrapType !== WrapType.NOT_APPLICABLE && wrapError) ||
       approvalState === ApproveOrPermitState.PENDING_SIGNATURE ||
-      !(inputCurrencyAmount && outputCurrencyAmount && inputCurrencyBalance) ||
-      (inputCurrencyAmount && inputCurrencyBalance && inputCurrencyBalance.lessThan(inputCurrencyAmount)),
-    [approvalState, chainId, disabled, inputCurrencyAmount, inputCurrencyBalance, outputCurrencyAmount]
+      !(inputTradeCurrencyAmount && inputCurrencyBalance) ||
+      (inputTradeCurrencyAmount && inputCurrencyBalance && inputCurrencyBalance.lessThan(inputTradeCurrencyAmount)),
```

Comment:
this is redundant with L101 - can you replace `(inputTradeCurrencyAmount && inputCurrencyBalance && inputCurrencyBalance.lessThan(inputTradeCurrencyAmount))` with `inputCurrencyBalance.lessThan(inputTradeCurrencyAmount)`?